index.bs

```diff
@@ -5452,6 +5456,59 @@ function is [=salted=] with [=salt=] values private to the [=[RP]=], since hashi
 values. It is RECOMMENDED to let the [=user handle=] be 64 random bytes, and store this value in the user's account.
 
 
+## Username Enumeration ## {#sctn-username-enumeration}
+
+While initiating a [=registration ceremony|registration=] or [=authentication ceremony=], there is a risk that the [=[WRP]=] might leak sensitive
+information about its registered users. For example, if a [=[RP]=] uses e-mail addresses as usernames and an attacker attempts to
+initiate an [=authentication=] [=ceremony=] for "alex.p.mueller@example.com" and the [=[RP]=] responds with a failure, but then
+successfully initiates an [=authentication ceremony=] for "j.doe@example.com", then the attacker can conclude that "j.doe@example.com"
+is registered and "alex.p.mueller@example.com" is not. The [=[RP]=] has thus leaked the possibly sensitive information that
+"j.doe@example.com" has an account at this [=[RP]=].
+
+The following is a non-normative, non-exhaustive list of measures the [=[RP]=] may implement to mitigate or prevent information
+leakage due to such an attack:
+
+- For [=registration ceremonies=]:
+
+    - If the [=[RP]=] uses [=[RP]=]-specific usernames to identify users:
+
+        - When initiating a [=registration ceremony=], disallow registration of usernames that are syntactically valid e-mail
+            addresses.
+
+            Note: The motivation for this suggestion is that in this case the [=[RP]=] probably has no choice but to fail the
+            [=registration ceremony=] if the user attempts to register a username that is already registered, and an information
+            leak might therefore be unavoidable. By disallowing e-mail addresses as usernames, the impact of the leakage can be
+            mitigated since it will be less likely that a user has the same username at this [=[RP]=] as at other [=[RPS]=].
```

Comment:
Yeah, those are good points, and I've been having thoughts along those lines too since I wrote that passage. I'd gladly delete that recommendation and replace it with something pointing out that an information leak might be unavoidable, so username-less login might be preferable. Thanks for digging!